tests/test_proto_compatibility.py

```diff
@@ -0,0 +1,170 @@
+import os
+import subprocess
+import time
+import logging
+import signal
+import uuid
+import tempfile
+from pathlib import Path
+
+import requests
+import pytest
+import mlflow
+from tests.helper_functions import get_safe_port
+
+_logger = logging.getLogger(__name__)
+
+
+def wait_until_tracking_server_ready(tracking_uri):
+    for _ in range(10):
```

Comment:
Do you want to add a preemptive initial sleep here since the server always takes a few seconds to get up and running (to prevent log spam for expected initial set of `ConnectionError` exceptions)?

Reply:
```suggestion
    time.sleep(5)
    for _ in range(10):
```

Reply:
Sounds good. 5 might be too long. How about 3?